### PR TITLE
feat:[CDS-85486]: fix for infra description variable resolution in shell script

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
This fix involves addition of infrastructure description in infra outcome so that it can be accessed through expressions in shell script.

Tests Done:
1. Running shell script step with description variable. 
2. Validated the same for azure infra as well.

Screenshot 2023-11-28 at 4 55 09 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56057)

### CODE REVIEW

**Comment by sourabh-awashti** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/AbstractInfrastructureTaskExecutableStep.java [hunk 1]
```diff
@@ -206,7 +206,7 @@ protected TaskRequestData obtainTaskInternal(Ambiance ambiance, Infrastructure i
 
     final InfrastructureOutcome infrastructureOutcome =
         infrastructureOutcomeProvider.getOutcome(ambiance, infrastructure, environmentOutcome, serviceOutcome,
-            ngAccess.getAccountIdentifier(), ngAccess.getOrgIdentifier(), ngAccess.getProjectIdentifier(), tags);
+            ngAccess.getAccountIdentifier(), ngAccess.getOrgIdentifier(), ngAccess.getProjectIdentifier(), tags, null);
```
[the remark this answers is not in the document]
Comment (sourabh-awashti) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/AbstractInfrastructureTaskExecutableStep.java:209]: this is for v1 infra step, and here we don't have information about description as we don't have infra entity. For V2 infra step, i have added the description

Comment (tchaurasiya) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/AbstractInfrastructureTaskExecutableStep.java:209]: This is getting used for winrm and ssh type of infra lets propagate description here too.

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/AbstractInfrastructureTaskExecutableStep.java (hunk 1 above)
Comment (sourabh-awashti) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/infra/steps/AbstractInfrastructureTaskExecutableStep.java:209]: done

**Comment by sourabh-awashti** [comment]
trigger messagecheck

**Review by tchaurasiya — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 8 of 11 files at r1, 3 of 4 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56057)* status: 13 file(s) to review. 2. unresolved discussions.

**Comment by sourabh-awashti** [comment]
trigger review

**Pull request merged**